src/store.cc

```diff
@@ -1442,8 +1442,11 @@ StoreEntry::updateOnNotModified(const StoreEntry &e304)
     // update reply before calling timestampsSet() below
     const auto &oldReply = mem_obj->freshestReply();
     const auto updatedReply = oldReply.recreateOnNotModified(e304.mem_obj->baseReply());
-    if (updatedReply) // HTTP 304 brought in new information
+    if (updatedReply) { // HTTP 304 brought in new information
+        if (updatedReply->header.tooLarge())
```

Comment:
If other Squid code includes response status line length when checking reply_header_max_size, then we should preserve that existing logic in this new code.  Doing so may require refactoring tooLarge() to become a virtual Http::Message method, so that HttpRequest and HttpReply can apply their own start line calculations and limits. However, if this is the only place where we expect to call tooLarge(), then perhaps we do not need that method at all and can do something much simpler instead? For example:

```suggestion
        if (updatedReply->expectedPackedSize() > Config.maxReplyHeaderSize)
```

... where packedSize() is a new HttpReply method, similar to the length() method added in this PR. HttpReply::packedSize() would sum up the packed status line size (another new method?) and HttpHeader::len.

Reply:
>  perhaps we do not need that method at all 

Agreed. I posted a couple of changes: 28dc754, introducing  HttpReply::prefixLen() and b7acd7, removing tooLarge(). I reused some old code from Http::One::ResponseParser and also adjusted the same method in HttpRequest.